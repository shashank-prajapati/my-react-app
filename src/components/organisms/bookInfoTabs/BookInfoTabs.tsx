import React from "react";
import Tabs from "@mui/material/Tabs";
import Tab from "@mui/material/Tab";
import Typography from "@mui/material/Typography";
import Box, { BoxProps } from "@mui/material/Box";
import TabPanel from "../../atoms/tabPanel/TabPanel";

function a11yProps(index: number) {
  return {
    id: `simple-tab-${index}`,
    "aria-controls": `simple-tabpanel-${index}`,
  };
}

function BookInfoTabs(props: BoxProps) {
  const { sx } = props;
  const [value, setValue] = React.useState(0);

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };

  return (
    <Box sx={{ width: "64%", ...sx, mt: 15 }}>
      <Box sx={{ borderBottom: 1, borderColor: "divider" }}>
        <Tabs
          value={value}
          onChange={handleChange}
          aria-label="basic tabs example"
          sx={{
            ".MuiTab-root": {
              alignItems: "flex-start",
              p: 0,
              width: "200px",
              textTransform: "none",
              "&.Mui-selected": {
                color: "navTextColors.main",
              },
              fontWeight: "bold",
            },
          }}
        >
          <Tab label="Synopsis" {...a11yProps(0)} />
          <Tab label="Who is it for?" {...a11yProps(1)} />
          <Tab label="About the author" {...a11yProps(2)} />
        </Tabs>
      </Box>
      <TabPanel value={value} index={0}>
        <Typography variant="body2" color="navTextColors.main">
          Beyond Entrepreneurship 2.0 (2020) updates Jim Collins and Bill
          Lazier’s essential 1992 business handbook, Beyond Entrepreneurship for
          the entrepreneurs, visionaries, and innovators of today. This new
          edition combines the timeless business advice and strategy of the
          original text, supplemented with cutting-edge insights and case
          studies pertinent to today’s business world.
        </Typography>
      </TabPanel>
      <TabPanel value={value} index={1}></TabPanel>
      <TabPanel value={value} index={2}></TabPanel>
    </Box>
  );
}

export default BookInfoTabs;
