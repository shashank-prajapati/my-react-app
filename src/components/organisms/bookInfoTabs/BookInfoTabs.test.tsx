import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import BookInfoTabs from "./BookInfoTabs";

describe("Testing Book Info Tabs", () => {
  it("should match the Book Info Tabs snapshot", () => {
    const { container } = render(<BookInfoTabs />);
    expect(container).toMatchSnapshot();
  });
});
