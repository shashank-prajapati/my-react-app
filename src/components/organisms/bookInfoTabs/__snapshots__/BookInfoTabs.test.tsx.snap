// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book Info Tabs should match the Book Info Tabs snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-o64vlo"
  >
    <div
      class="MuiBox-root css-1gsv261"
    >
      <div
        class="MuiTabs-root css-443yoj-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            aria-label="basic tabs example"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-controls="simple-tabpanel-0"
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Synopsis
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="simple-tabpanel-1"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Who is it for?
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="simple-tabpanel-2"
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
              id="simple-tab-2"
              role="tab"
              tabindex="-1"
              type="button"
            >
              About the author
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-0"
      id="simple-tabpanel-0"
      role="tabpanel"
    >
      <div
        class="MuiBox-root css-pxpt32"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-1amsg4u-MuiTypography-root"
        >
          Beyond Entrepreneurship 2.0 (2020) updates Jim Collins and Bill Lazier’s essential 1992 business handbook, Beyond Entrepreneurship for the entrepreneurs, visionaries, and innovators of today. This new edition combines the timeless business advice and strategy of the original text, supplemented with cutting-edge insights and case studies pertinent to today’s business world.
        </p>
      </div>
    </div>
    <div
      aria-labelledby="simple-tab-1"
      hidden=""
      id="simple-tabpanel-1"
      role="tabpanel"
    />
    <div
      aria-labelledby="simple-tab-2"
      hidden=""
      id="simple-tabpanel-2"
      role="tabpanel"
    />
  </div>
</div>
`;
